src/main/java/gregtech/api/unification/material/properties/FluidProperty.java

```diff
@@ -33,4 +39,19 @@ public void setPrimaryKey(@Nullable FluidStorageKey primaryKey) {
 
     @Override
     public void verifyProperty(MaterialProperties properties) {}
+
+    public Fluid solidifiesFrom(Material material) {
+        if (this.solidifyingFluid == null) {
+            return material.getFluid(FluidStorageKeys.LIQUID);
```

Comment:
This should not take in a `material`. You can retrieve the fluid with a key using methods in the `storage` object already in this class.